lms/static/js/learner_analytics_dashboard/LearnerAnalyticsDashboard.jsx

```diff
@@ -0,0 +1,145 @@
+/* global gettext */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import CircleChart from './CircleChart';
+import CircleChartLegend from './CircleChartLegend';
+import GradeTable from './GradeTable';
+import DueDates from './DueDates';
+import Discussions from './Discussions';
+
+function arrayToObject(array) {
+  return array.reduce((accumulator, obj) => {
+    const key = Object.keys(obj)[0];
+    accumulator[key] = obj[key];
+    return accumulator;
+  }, {})
+}
+
+function countByType(type, assignments) {
+  let count = 0;
+  assignments.map(({format}) => {
+    if (format === type) {
+      count += 1;
+    }
+  })
+  return count;
+}
+
+function getActiveUserString(count) {
+  const users = (count === 1) ? 'User' : 'Users';
+  return `${users} active in this course right now`;
+}
+
+function getAssignmentCounts(types, assignments) {
+  const countsArray = types.map((type) => {
+    return {
+      [type]: countByType(type, assignments)
+    }
+  });
+
+  return arrayToObject(countsArray);
+}
+
+function getStreakIcons(count) {
+  return Array.apply(null, { length: count }).map((e, i) => (
+    <span className="fa fa-trophy" aria-hidden="true" key={i}></span>
+  ));
+}
+
+function getStreakString(count) {
+  const unit = (count ===1) ? 'week' : 'weeks';
+  return `Logged in ${count} ${unit} in a row`;
+}
+
+export function LearnerAnalyticsDashboard(props) {
+console.log('props: ', props);
```

Comment:
Should we remove this `console` statement?

Reply:
Good catch.